packages/devtools_app/lib/src/auto_dispose.dart

```diff
@@ -47,7 +47,13 @@ class Disposer {
 
     assert(_listenables.length == _listeners.length);
     for (int i = 0; i < _listenables.length; ++i) {
-      _listenables[i].removeListener(_listeners[i]);
+      try {
+        _listenables[i].removeListener(_listeners[i]);
+      } on FlutterError catch (e) {
+        // If the listenable has already been disposed, it will throw an error
```

Comment:
we should catch the underlying error instead of hiding it with a try catch block.

Reply:
can we fix the order of execution instead so that the listeners are removed before the listenables are disposed?

Reply:
I like both of these ideas a lot.

> we should catch the underlying error instead of hiding it with a try catch block.
The underlying error can change.  In debug mode it will be a FlutterError thrown by an assert.  In release mode, it will be a null pointer error because the underlying notifier's _listeners list is null.

> can we fix the order of execution instead so that the listeners are removed before the listenables are disposed?

We can by overriding the InheritedElement behavior.  I originally felt like that was going to be a bit too much boilerplate in the Controllers class, and it will be a few lines more of code than the try/catch.  However, having slept on it I feel a lot better about delaying the disposal than I feel about this try/catch.

Reply:
Done.